api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -88,7 +107,22 @@ public String getIdentifierForSeed(long seed) {
     	return identifier;
     }
 
-    //***** PROPERTY ACCESS *****
+	private String getPrefixProviderBeanNameIfApplicable(String prefix) {
```

Comment:
Why don't you just get the `PrefixProvider` directly, returning null if none applicable or even returning a simple (new) implementation that provides the constant prefix when applicable?
Something like:
```java
private PrefixProvider getPrefixProvider(String prefix) { ...
```

And _of course_, could you provide a bunch of unit tests for this? Using mocks.